From dea2ddb038b01eff16f5da4a19c7282e438ec19a Mon Sep 17 00:00:00 2001
From: Jonathan Steffan <jsteffan@fedoraproject.org>
Date: Sat, 8 Jul 2023 11:50:01 -0600
Subject: [PATCH] Add Python 3.12 Support
Forwarded: https://github.com/graphite-project/carbon/pull/951

 - Remove use of imp module (setKeyFunction*)
 - Update tests for unittest changes
 - Update ConfigParser read
---
 lib/carbon/routers.py              | 12 ------------
 lib/carbon/tests/test_log.py       |  2 +-
 lib/carbon/tests/test_protocols.py |  8 ++++----
 lib/carbon/tests/test_routers.py   |  6 +++---
 lib/carbon/tests/test_storage.py   | 26 +++++++++++++-------------
 lib/carbon/tests/test_util.py      | 16 ++++++++--------
 setup.py                           |  2 +-
 7 files changed, 30 insertions(+), 42 deletions(-)

--- graphite-carbon.orig/lib/carbon/routers.py	2024-07-05 14:55:45.369827038 +0200
+++ graphite-carbon/lib/carbon/routers.py	2024-07-05 14:55:45.365826955 +0200
@@ -1,4 +1,3 @@
-import imp
 from carbon.hashing import ConsistentHashRing, carbonHash
 from carbon.util import PluginRegistrar
 from six import with_metaclass
@@ -125,17 +124,6 @@
   def getKey(self, metric):
     return metric
 
-  def setKeyFunction(self, func):
-    self.getKey = func
-
-  def setKeyFunctionFromModule(self, keyfunc_spec):
-    module_path, func_name = keyfunc_spec.rsplit(':', 1)
-    module_file = open(module_path, 'U')
-    description = ('.py', 'U', imp.PY_SOURCE)
-    module = imp.load_module('keyfunc_module', module_file, module_path, description)
-    keyfunc = getattr(module, func_name)
-    self.setKeyFunction(keyfunc)
-
 
 class AggregatedConsistentHashingRouter(DatapointRouter):
   plugin_name = 'aggregated-consistent-hashing'
--- graphite-carbon.orig/setup.py	2024-07-05 14:55:45.369827038 +0200
+++ graphite-carbon/setup.py	2024-07-05 14:55:45.365826955 +0200
@@ -24,7 +24,7 @@
 with open('setup.cfg', 'r') as f:
     orig_setup_cfg = f.read()
     f.seek(0)
-    cf.readfp(f, 'setup.cfg')
+    cf.read_file(f, 'setup.cfg')
 
 if os.environ.get('GRAPHITE_NO_PREFIX'):
     cf.remove_section('install')
